imap_processing/spice/geometry.py

```diff
@@ -149,13 +149,45 @@ def get_spacecraft_spin_phase(
     spin_phase : float or np.ndarray
         Spin phase for the input query times.
     """
+    spin_df = get_spin_data()
+
+    # Combine spin_start_sec and spin_start_subsec to get the spin start
+    # time in seconds. The spin start subseconds are in milliseconds.
+    # TODO: Decide if we should do this calculation in the data itself
```

Comment:
It makes sense to me to add a floating point MET column in the `get_spin_data()` function.